server/sonar-web/src/main/js/apps/permissions/global/store/actions.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as api from '../../../../api/permissions';
+import { parseError } from '../../../code/utils';
+import { raiseError } from '../../shared/store/actions';
+import { getQuery, getFilter } from '../../shared/store/rootReducer';
+
+export const loadHolders = () => (dispatch, getState) => {
+  const query = getQuery(getState());
+  const filter = getFilter(getState());
+
+  dispatch({ type: 'REQUEST_HOLDERS', query });
+
+  const requests = [];
+
+  if (filter !== 'groups') {
+    requests.push(api.getGlobalPermissionsUsers(query));
+  } else {
+    requests.push(Promise.resolve([]));
+  }
+
+  if (filter !== 'users') {
+    requests.push(api.getGlobalPermissionsGroups(query));
+  } else {
+    requests.push(Promise.resolve([]));
+  }
+
+  return Promise.all(requests).then(responses => (
+      dispatch({
+        type: 'RECEIVE_HOLDERS_SUCCESS',
+        users: responses[0],
+        groups: responses[1],
+        query
+      })
+  )).catch(e => {
+    return parseError(e).then(message => dispatch(raiseError(message)));
+  });
+};
+
+export const updateQuery = (query = '') => dispatch => {
+  dispatch({ type: 'UPDATE_QUERY', query });
+  if (query.length === 0 || query.length > 2) {
+    dispatch(loadHolders());
+  }
+};
+
+export const updateFilter = filter => dispatch => {
+  dispatch({ type: 'UPDATE_FILTER', filter });
+  dispatch(loadHolders());
+};
+
+export const grantToUser = (login, permission) => dispatch => {
+  api.grantPermissionToUser(null, login, permission).then(() => {
+    dispatch({ type: 'GRANT_PERMISSION_TO_USER', login, permission });
+  }).catch(e => {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Remove curly braces and "return" from this arrow function body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)